libraries/chain/stake.cpp

```diff
@@ -139,12 +146,15 @@ void recall_proxied(cyberway::chaindb::chaindb_controller& db, const cyberway::c
 
     auto agents_table = db.get_table<stake_agent_object>();
     auto grants_table = db.get_table<stake_grant_object>();
+    auto autorcs_table = db.get_table<stake_auto_recall_object>();
     auto agents_idx = agents_table.get_index<stake_agent_object::by_key>();
     auto grants_idx = grants_table.get_index<stake_grant_object::by_key>();
+    auto autorcs_idx = autorcs_table.get_index<stake_auto_recall_object::by_key>();
 
     auto grantor_as_agent = get_agent(token_code, agents_idx, grantor_name);
     std::map<account_name, int64_t> votes_changes;
-    update_proxied_traversal(storage, now, token_code, agents_idx, grants_idx, grantor_as_agent, time_point_sec(), votes_changes, true);
+    update_proxied_traversal(storage, now, token_code, agents_idx, grants_idx, autorcs_idx, grantor_as_agent, time_point_sec(), votes_changes, true,
+        autorcs_idx.find(agent_key(token_code, account_name{})) != autorcs_idx.end());
```

Comment:
At the current time this will require the full replay.
To avoid the full replay we can add verifying of system tables on each start of the nodeos.

Reply:
System contract is initialized in this place:
https://github.com/cyberway/cyberway/blob/master/libraries/chain/controller.cpp#L572

Reply:
At this place https://github.com/cyberway/cyberway/blob/master/libraries/chain/controller.cpp#L468 the structure of system contract is loaded from the current state - it can be upgraded via setabi.
But there is no way to check that table is exist.